File: app/src/main/java/net/north101/android/ghplayertracker/data/Elements.java
package net.north101.android.ghplayertracker.data;

import org.json.JSONException;

public enum Elements {
    fire,
    ice,
    wind,
    earth,
    sun,
    moon;

    public static Elements parse(String elementData) throws JSONException {
        switch (elementData) {
            case "fire":
                return Elements.fire;

            case "ice":
                return Elements.ice;

            case "earth":
                return Elements.earth;

            case "wind":
                return Elements.wind;

            case "sun":
                return Elements.sun;

            case "moon":
                return Elements.moon;

            default:
                throw new JSONException("");
        }
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/Elements.kt
package net.north101.android.ghplayertracker.data

import org.json.JSONException

enum class Elements {
    fire,
    ice,
    air,
    earth,
    light,
    dark;

    companion object {
        @Throws(JSONException::class)
        fun parse(elementData: String): Elements {
            return when (elementData) {
                "fire" -> Elements.fire
                "ice" -> Elements.ice
                "earth" -> Elements.earth
                "air" -> Elements.air
                "light" -> Elements.light
                "dark" -> Elements.dark
                else -> throw JSONException(elementData)
            }
        }
    }
}
